program/tests/tests/add_remove_validator.rs

```diff
@@ -34,24 +38,30 @@ async fn test_successful_add_validator() {
 }
 
 #[tokio::test]
-async fn test_remove_validator_without_unclaimed_credits() {
+async fn test_successful_remove_validator() {
     let mut context = Context::new_with_maintainer_and_validator().await;
-
-    let solido = context.get_solido().await;
-    assert_eq!(solido.validators.len(), 1);
-
-    let vote_account = solido.validators.entries[0].pubkey;
-    assert_eq!(solido.validators.entries[0].entry.fee_credit, StLamports(0));
-
+    let validator = &context.get_solido().await.validators.entries[0];
+    context.deactivate_validator(validator.pubkey).await;
     context
-        .try_remove_validator(vote_account)
+        .try_remove_validator(validator.pubkey)
         .await
-        .expect("Failed to remove validator.");
+        .unwrap();
 
     let solido = context.get_solido().await;
     assert_eq!(solido.validators.len(), 0);
 }
 
+#[tokio::test]
+async fn test_removing_validator_with_stake_accounts_should_fail() {
+    let (mut context, _) = Context::new_with_two_stake_accounts().await;
+    let validator = &context.get_solido().await.validators.entries[0];
+    let result = context.try_remove_validator(validator.pubkey).await;
+
+    // The validator should not be able to be removed if it is still active
+    //  (i.e. the active flag is set toe true OR it has stake accounts)
+    assert_solido_error!(result, LidoError::ValidatorIsStillActive);
+}
+
```

Comment:
We should also add a third test where the validator has no stake accounts, but it does have unclaimed credits, and test that we refuse to remove it in that case too.